crates/flux-refineck/src/type_env/place_ty.rs

```diff
@@ -118,9 +126,14 @@ impl LookupMode for NoUnfold {
 }
 
 impl PlacesTree {
-    pub(crate) fn unfold(&mut self, infcx: &mut InferCtxt, key: &impl LookupKey) -> InferResult {
+    pub(crate) fn unfold(
+        &mut self,
+        infcx: &mut InferCtxt,
+        key: &impl LookupKey,
+        span: Span,
+    ) -> InferResult {
```

Comment:
Not sure how well this would work, but instead of adding a `Span` to all these functions, we could try passing an `InferCtxtAt` directly

Reply:
Ok, but there's some simpler cleanup we can do. Instead of passing a `Span` to `lookup_inner` and then threading it down, we can put the `Span` in the `Unfold` struct. Then we don't need to pass a `Span` when there's no `InferCtxt`.